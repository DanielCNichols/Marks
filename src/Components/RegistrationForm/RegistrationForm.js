import React, { useState } from 'react';
import {
  useRegistrationForm,
  validationRules,
} from '../../Hooks/useRegistrationForm';
import AuthApiService from '../../Services/authService';
import TokenService from '../../Services/tokenService';

export default function RegistrationForm(props) {
  const [error, setError] = useState(null);
  const {
    inputs,
    handleChange,
    handleSubmit,
    inputErrors,
  } = useRegistrationForm(handleRegister, validationRules);

  async function handleRegister(inputs) {
    try {
      await AuthApiService.postUser(inputs);
      props.history.push('/login');
    } catch (error) {
      setError(error);
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <fieldset>
        <legend>Sign Up</legend>
        <div className="form-element">
          <label htmlFor="username">Username</label>
          <input
            type="text"
            name="username"
            id="username"
            onChange={handleChange}
            value={inputs.username}
          />
        </div>
        <div className="form-element">
          <label htmlFor="password">Password</label>
          <input
            type="text"
            name="password"
            id="password"
            onChange={handleChange}
            value={inputs.password}
          />
        </div>
        <div className="form-element">
          <label htmlFor="confirmPass">Password</label>
          <input
            type="text"
            name="confirmPass"
            id="confirmPass"
            onChange={handleChange}
            value={inputs.confirmPass}
          />
        </div>

        {inputErrors && (
          <>
            {Object.keys(inputErrors).map((e, idx) => {
              return (
                <p style={{ color: 'white' }} key={idx}>
                  {inputErrors[e]}
                </p>
              );
            })}
          </>
        )}

        {error && (
          <>
            {Object.keys(error).map((e, idx) => {
              return (
                <p style={{ color: 'white' }} key={idx}>
                  {error[e]}
                </p>
              );
            })}
          </>
        )}

        {/* {error && <p className="error">{error}</p>} */}

        <button type="submit">Sign up</button>
      </fieldset>
    </form>
  );
}
