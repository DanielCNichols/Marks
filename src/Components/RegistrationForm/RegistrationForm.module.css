@import '../../vars.module.css';

.error,
.postError {
  color: red;
  text-align: left;
  font-size: 0.8em;
}

.postError {
  margin: 0 auto 10px auto;
  width: 90%;
}

.registrationForm {
  margin: 0 auto;
  width: 100%;
  border: 1px solid rgba(255, 255, 255, 0.38);
  border-radius: 6px;
  padding: 20px 10px;
}

.registrationForm fieldset,
.formElement {
  display: flex;
  flex-direction: column;
  width: 90%;
  margin: auto;
}

.registrationForm legend {
  text-align: center;
  color: var(--color-subheading);
  margin-bottom: 20px;
  font-weight: bold;
}

.formElement {
  margin-bottom: 10px;
}

.formElement input {
  border-radius: 3px;
  height: 30px;
  padding: 3px 5px;
}

.formElement label {
  margin-bottom: 5px;
}

.submit {
  display: block;
  width: 50%;
  max-width: 150px;
  height: 40px;
  padding: 10px;
  margin: 0 auto;
  background-color: #23f84f;
  border-radius: 5px;
  color: black;
  margin-bottom: 20px;
}

.loginTag {
  color: var(--color-subheading);
  font-size: 0.8em;
  text-align: center;
}

.loginTag a {
  color: white;
}

/* tablet styles */

/* Desktop Styles */

@media screen and (min-width: 800px) {
  .registrationForm {
    width: 80%;
  }
}
