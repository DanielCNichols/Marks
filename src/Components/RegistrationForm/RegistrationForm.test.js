import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RegistrationForm from './RegistrationForm';
import { BrowserRouter } from 'react-router-dom';

describe('Registration Form tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <RegistrationForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <RegistrationForm />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
