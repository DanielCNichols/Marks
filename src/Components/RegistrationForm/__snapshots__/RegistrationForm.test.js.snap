// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form tests renders the UI as expected 1`] = `
<form
  className="registrationForm"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Sign Up
    </legend>
    <div
      className="formElement"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        aria-placeholder="markit146"
        aria-required="true"
        autoComplete="username"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="markit146"
        type="text"
        value=""
      />
    </div>
    <div
      className="formElement"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        aria-placeholder="Pass123"
        aria-required="true"
        autoComplete="password"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Pass123"
        type="password"
        value=""
      />
    </div>
    <div
      className="formElement"
    >
      <label
        htmlFor="confirmPass"
      >
        Confirm Password
      </label>
      <input
        aria-placeholder="Pass123"
        aria-required="true"
        autoComplete="confirmPass"
        id="confirmPass"
        name="confirmPass"
        onChange={[Function]}
        placeholder="Pass123"
        type="password"
        value=""
      />
    </div>
    <button
      className="submit"
      type="submit"
    >
      Sign up
    </button>
  </fieldset>
  <p
    className="loginTag"
  >
    Already have an account? 
    <a
      href="/login"
      onClick={[Function]}
    >
      Log in!
    </a>
  </p>
</form>
`;
